### PR TITLE
Adds pre-commit hook that will get installed on npm install. This will run linters

### PR DESCRIPTION
**Pull request by rcorral**

### CODE REVIEW

File: .pre-commit-hook [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+set -e
+
+echo "Pre-commit checks..."
+
+npm run lint;
+
+if [ "undefined" != `npm config get externalplugins` ]; then
+  npm run lint-plugins;
+fi
```
Comment (jfurrow) [inline .pre-commit-hook]: This needs more emoji...

**Comment by jfurrow** [comment]
This is pretty sweet! It might get annoying when creating several commits from a bunch of changes, but it's so much better than the alternative.

File: .pre-commit-hook [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+set -e
+
+echo "Pre-commit checks..."
```
Comment (jfurrow) [inline .pre-commit-hook]: This needs more emoji...

Comment (rcorral) [inline .pre-commit-hook]: Done!

**Pull request merged**